lib/mutant/matcher/config.rb

```diff
@@ -84,6 +84,10 @@ def merge(other)
         )
       end
 
+      def relevant_path?(source_path)
+        subject_filters.all? { |filter| filter.modified_path?(source_path) }
```

Comment:
The contract for subject filters so far is `#call` and I've got mutant clients that have custom subject filters that this code will fail if we where to merge it.

IHMO we should move the repository diff to a core concept in the matcher config allowing to directly reference it without having "all" subject filters having to support this specific concept.

Reply:
Ok, that makes sense. Would you like me to change something specifically here? Or how should we proceed?

Reply:
I've done the change for you.